src/components/header.jsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+
+const propTypes = {
+  title: React.PropTypes.string
+};
+
+export default class HeaderComponent extends React.Component {
+
+  //This constructor fires on the instantiation of the component, props is passed in, and we HAVE to super(props) to specify THIS instance (its weird I know), then we can set the state property directly to equal a new object. Equivelant would be getInitialState
+  constructor(props) {
+    super(props);
+    this.state = {
+      header: this.props.title
+    }
```

Comment:
Over here, the 
```js
this.state = {header: this.props.title}
```
Is essentially doing the same thing as:

```js
this.setState({header: this.props.title}).
```

Normally, you run that line inside of the method, like this

```js
getInitialState: function(){
return {header: this.props.title}.
}
```

except we don't need to do that with es2015 class notation, because we have something called the constructor which fires whenever a class is 'constructed' - getInitialState has roughly the same lifecycle, and all it does is set the state property on 'this' anyway. In the constructor, you can directly access the state, because it runs once, and at the end react does it's first render automatically. However, in any subsequent modifications of the state property, it needs to be made through the **this.setState({prop: newVal})** notation